build/shade/EnvironmentParameters.shade

```diff
@@ -0,0 +1,20 @@
+use namespace='System'
+ 
+functions @{
+    static string GetEnvironmentParameter(string name)
+    {
+        return GetEnvironmentParameter(name, s => s);
+    } 
+
+    static T GetEnvironmentParameter<T>(string name, Func<string, T> valueFn)
+    {
+        var value = valueFn(Environment.GetEnvironmentVariable(name));
+        Console.WriteLine(name + ": " + (value != null ? value.ToString() : "(null)"));
+        return value;
+    }
+
+    static Func<string, string> DefaultTo(string defaultValue)
```

Comment:
Why would I use this over just `v ?? default`?

One of the concerns I have is that it is hard to find the definition of a Sake function/target and there is no a "go to reference"-like functionality. So if I saw "ToDefault" I would have to find the definition first to understand the script.   Figuring out bugs in Sake scripts is no fun and this in my mind may make it worse.